fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/QueryPlan.java

```diff
@@ -128,6 +129,15 @@
      */
     boolean hasLoadBykeys();
 
+    /**
+     * Indicates whether this plan can only return one or zero records.
+     * @param metaData meta-data to use to determine index uniqueness
+     * @return <code>true</code> if this plan returns at most a single record.
+     */
+    default boolean isUnique(@Nonnull RecordMetaData metaData) {
```

Comment:
The name feels like it's misleading as it is not about uniqueness but about maximum cardinality. Other planners use the name maxCardinality which can be `UNKNOWN` and a real number such as `0`, `1`. Where you would mostly expect `1` or `UNKNOWN`. But even things like other actual numbers can be useful, i.e. for understanding the union of three arms of a `UNION` of `maxCardinality == 1` arms would be `3`.

Reply:
Changed to `maxCardinality`.